src/plugins/data/public/query/timefilter/timefilter.ts

```diff
@@ -156,12 +162,18 @@ export class Timefilter {
     if (prevRefreshInterval?.value > 0 && newRefreshInterval.value <= 0) {
       shouldUnpauseRefreshLoop = false;
     }
+
+    this._isRefreshIntervalTouched =
+      this._isRefreshIntervalTouched ||
+      areRefreshIntervalsDifferent(this.refreshIntervalDefaults, newRefreshInterval);
```

Comment:
should we consider refresh interval touched even when somebody sets the value to the same value the defaults are set to ?
this would mean we need to introduce resetRefreshInterval method however. cc @lukasolson